src/data/fin/succ_pred.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2022 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import order.succ_pred.basic
+
+/-!
+# Successors and predecessors of `fin n`
+
+In this file, we show that `fin n` is both a `succ_order` and a `pred_order`. Note that they are
+also archimidean, but this is derived from the general instance for well-orderings as opposed
+to a specific `fin` instance.
+
+-/
+
+namespace fin
+
+instance : ∀ {n : ℕ}, succ_order (fin n)
+| 0 := by constructor; exact elim0
+| (n+1) :=
+_root_.succ_order.of_core (λ i, if i < fin.last n then i + 1 else i)
```

Comment:
```suggestion
succ_order.of_core (λ i, if i < fin.last n then i + 1 else i)
```

Reply:
no, the equation compiler term created is `fin.succ_order` and it tries to prefer that. so this is necessary

Reply:
Uh, okay, wasn't expecting that 😆